db/primary_migrate/20240110141229_add_disposable_email_domains_table.rb

```diff
@@ -0,0 +1,11 @@
+class AddDisposableEmailDomainsTable < ActiveRecord::Migration[7.1]
+  def change
```

Comment:
I had considered doing something like this, but (1) it included a few additional unexpected metadata in `schema.rb` , (2) it would prevent automatic rollbacks, and (3) the table itself is rather simple, so copying the original migration seemed reasonable.

```rb
def change
  safety_assured do
    execute <<-SQL
      CREATE TABLE disposable_email_domains (LIKE disposable_domains INCLUDING ALL);
    SQL
  end
end
```

I had also considered doing this in a way where we create an "alias" of sorts for the old table name, which might have allowed this to be a two-step process. But I wasn't sure how safe this would be to perform in a single migration.

```rb
def change
  safety_assured do
    rename_table :disposable_domains, :disposable_email_domains
    execute <<-SQL
      CREATE VIEW disposable_domains AS SELECT * FROM disposable_email_domains;
    SQL
  end
end
```

Reply:
Another option would be to raw SQL those two commands to be certain they happen in a single transaction. If this were for a more used table I'd strongly recommend that